app/utils/config.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+"""apps.utils.config: imports and finalize config.yml"""
+
+from pathlib import Path
+import glob
+import yaml
+import os
+
+class PyVar(yaml.YAMLObject):
+    """
+    a class to create a custom !PY tag in yaml
+    The !PY tag allows calling python code stored in a dictionary
+    """
+    # this is a PYYAML requirment and is case sensitive
+    yaml_tag = u'!PY'
+
+    def __init__(self):
+        """
+        starts with empty dictionary
+        """
+        self.py_dict = {}
+
+    def get(self,key):
+        """
+        retrieves a value from the dictionary
+        """
+        return self.py_dict[key]
+
+    def add(self,key,value):
+        """
+        adds a new key value pair to a dictionary
+        """
+        self.py_dict[key] = value
+
+
+# construct all they key value pairs that can be referenced by the yaml file
+# creating a function like this is required for yaml.SafeLoader.add_constructor
+def yaml_construct(loader,node):
+    c = PyVar()
+    ## start adding python variables here
+    c.add('base_dir',Path(__file__).resolve(strict=True).parent.parent.parent.absolute())
+    c.add('app_dir', c.get('base_dir') / 'app')
+    c.add('data_dir', c.get('app_dir') / 'data')
+    c.add('log_dir', c.get('app_dir') / 'log')
+    c.add('json_file_name',os.path.basename(max(glob.glob(str(c.get('data_dir')) + '/*.json'), key=os.path.getctime)))
```

Comment:
I have a drugs-ndc.json in the data folder however the following error occurs:
```
File "app/utils/config.py", line 45, in yaml_construct
    c.add('json_file_name',os.path.basename(max(glob.glob(str(c.get('data_dir')) + '/*.json'), key=os.path.getctime)))
ValueError: max() arg is an empty sequence```

Reply:
> @hl1 - Currently, the code does not use any of these variables. This is the first step, where you have created a script to capture the YAML information. Correct?
> 
> Also, please be sure to update the requirements with YAML and specify the latest version number.

Correct. this is an independent system that could be added to master without impacting anything. integration is coming.

Reply:
> I have a drugs-ndc.json in the data folder however the following error occurs:
> 
> ```
> File "app/utils/config.py", line 45, in yaml_construct
>     c.add('json_file_name',os.path.basename(max(glob.glob(str(c.get('data_dir')) + '/*.json'), key=os.path.getctime)))
> ValueError: max() arg is an empty sequence```
> ```

hmmm... is the data folder in `/app/data` or `/data`?